### PR TITLE
Some improvements to plugin initialization

### PR DESCRIPTION
**Pull request by bdm-k**
### Changes
* Fixed vulnerability in dependencies by running `npm audit fix`.
* Removed redundant `sortSnippets` calls in "main.ts". Snippets returned by the function `parseSnippets` are already sorted.
* Refactored the function `refreshCMExtensions` and renamed it to `setEditorExtensions`.
  * Moved the line calling `this.app.workspace.updateOptions` outside of the function so that its side effect is limited to modifying `this.editorExtensions`.
  * Eliminated the use of `latexSuiteExtensions` and `optionalExtensions`. The previous code feels hacky because it required `@ts-ignore`. In addition, I found the name `latexSuiteExtensions` unclear.
* Removed the call to `reconfigureLatexSuiteConfig` in `processSettings`. The extensions reconfigured by `reconfigureLatexSuiteConfig` will be removed in the call to `refreshCMExtensions` (`setEditorExtensions` in new code), so `reconfigureLatexSuiteConfig` is unnecessary.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -141,13 +143,14 @@ export default class LatexSuitePlugin extends Plugin {
...
 
-		return sortSnippets(snippets);
+		return snippets;
 	}
 
 	async processSettings(becauseFileLocationUpdated = false, becauseFileUpdated = false) {
 		this.CMSettings = processLatexSuiteSettings(await this.getSnippets(becauseFileLocationUpdated, becauseFileUpdated), this.settings);
 		this.reconfigureLatexSuiteConfig();
 		this.refreshCMExtensions();
+		this.app.workspace.updateOptions();
```
Comment (bdm-k) [inline src/main.ts:153]: Moved this line outside of `refreshCMExtensions` to clarify the function's role.

Comment (bdm-k) [reply src/main.ts:153]: Maybe it is better to rename `refreshCMExtensions`.

**Comment by artisticat1** [comment]
Hi, thanks for the PR! Will try to review it soon.

File: src/main.ts [hunk 2]
```diff
@@ -141,41 +142,44 @@ export default class LatexSuitePlugin extends Plugin {
 
 		this.showSnippetsLoadedNotice(snippets.length, Object.keys(snippetVariables).length,  becauseFileLocationUpdated, becauseFileUpdated);
 
-		return sortSnippets(snippets);
+		return snippets;
```
Comment (artisticat1) [inline src/main.ts:145]: I believe it's necessary to keep the call to `sortSnippets()` here. When loading snippets from files, sorting needs to be done on the complete set of snippets from all snippet files — i.e. after all of the files have been loaded. `parseSnippets()` only sorts snippets _within_ files.

**Comment by bdm-k** [comment]
The commit history has become a bit messy. Should I clean it up and force push?

**Comment by artisticat1** [comment]
Sure, go ahead.

**Comment by bdm-k** [comment]
I'm done!

**Comment by artisticat1** [comment]
Thanks!

**Pull request merged**